src/Meshes/mesh.ts

```diff
@@ -4291,11 +4292,52 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
         subdivideWithSubMeshes?: boolean,
         multiMultiMaterials?: boolean
     ): Nullable<Mesh> {
+        let mergedMesh: Nullable<Mesh> = null;
+        Mesh._MergeMeshes(meshes, disposeSource, allow32BitsIndices, meshSubclass, subdivideWithSubMeshes, multiMultiMaterials, false, (mesh: Nullable<Mesh>) => mergedMesh = mesh);
+        return mergedMesh;
+    }
+
+    /**
+     * Merge the array of meshes into a single mesh for performance reasons.
+     * @param meshes defines he vertices source.  They should all be of the same material.  Entries can empty
+     * @param disposeSource when true (default), dispose of the vertices from the source meshes
+     * @param allow32BitsIndices when the sum of the vertices > 64k, this must be set to true
+     * @param meshSubclass when set, vertices inserted into this Mesh.  Meshes can then be merged into a Mesh sub-class.
+     * @param subdivideWithSubMeshes when true (false default), subdivide mesh to his subMesh array with meshes source.
+     * @param multiMultiMaterials when true (false default), subdivide mesh and accept multiple multi materials, ignores subdivideWithSubMeshes.
+     * @returns a new mesh
+     */
+    public static async MergeMeshesAsync(
+        meshes: Array<Mesh>,
+        disposeSource = true,
+        allow32BitsIndices?: boolean,
+        meshSubclass?: Mesh,
+        subdivideWithSubMeshes?: boolean,
+        multiMultiMaterials?: boolean
+    ): Promise<Nullable<Mesh>> {
+        let mergedMesh: Nullable<Mesh> = null;
+        await Mesh._MergeMeshes(meshes, disposeSource, allow32BitsIndices, meshSubclass, subdivideWithSubMeshes, multiMultiMaterials, true, (mesh: Nullable<Mesh>) => mergedMesh = mesh);
+        return mergedMesh;
+    }
+
+    private static async _MergeMeshes(
+        meshes: Array<Mesh>,
+        disposeSource = true,
+        allow32BitsIndices: boolean | undefined,
+        meshSubclass: Mesh | undefined,
+        subdivideWithSubMeshes: boolean | undefined,
+        multiMultiMaterials: boolean | undefined,
+        asynchronous: boolean,
+        onFinished: (mesh: Nullable<Mesh>) => void,
+    ): Promise<void> {
+        const quantizer = asynchronous ? createWorkQuantizer() : createWorkQuantizer(false);

```

Comment:
```suggestion
        const quantizer = createWorkQuantizer(!asynchronous);

```

Reply:
As written, `true` is not a valid argument for `createWorkQuantizer`. Currently you have to either pass false (meaning don't quantize), or a `number` (meaning quantize into time chunks of the specified size), or `undefined` (meaning do the default thing, which is quantizing to the default time chunk). I did it this way to try to make it more obvious what the resulting behavior would be. I feel like making `true` a valid argument is less clear, but if others feel strongly otherwise I will change it.